----------------------

<h3 style="color:blue;"><B>Question #1</b></h3>
<p style="color:DodgerBlue;font-size:12px">The gross profit of an item sold is the difference between its selling price and its purchase price. Two other marketing terms are: (25 marks)</p>
<p style="color:SlateBlue;font-size:12px">percentage gross profit = (gross profit) / (purchase price); and<br>profit margin = (gross profit) / (selling price)<br></p>
<p style="color:DodgerBlue;font-size:12px">where the percentage gross profit and profit margin are expressed as percentages. Write a program that asked an user for the selling prince and its purchase price on an item sold. Computes the gross profit, percentage gross profit, and profit margin of the item sold. Use functions for the input, compute percentage gross profit, compute profit margin, and output of the results.</p>

----------------------------

In [1]:
## The main program - i.e. control body
def q1Main():
    ## Call function inputPrices() to input the two prices
    ....

    ## Determine the gross profit
    ....

    ## Call function perFunction() to determine percentage markup
    ....

    ## Call function perFunction to determine profit margin
    ....

    ## Call the displayResult to displace the results
    ....

    
## Function to input the selling price and purchase price
def inputPrices():
    ....
    return (....)          # return the input selling price & purchase price


## Function return the percentage of an financial formula
def perFunction(....):      # pass two values via the parameters
    ....
    return (....)           # return the calculated percentage
    

## Function to display the results
def displayResults(....):  # pass all values to display via the parameters
    ....
    

q1Main()


Please enter the product's selling price : 6480
Please enter the product's purchase price: 3248

The selling price           : $  6,480.00
The purchase price          : $  3,248.00
The gross profit            : $  3,232.00
The percentage gross profit :       99.51%
The profit margin           :       49.88%


---------------------

<h3 style="color:blue;"><b>Question #2</b></h3>
<p style="color:DodgerBlue;font-size:12px;">Modify the program in question #1 for a number of users until a certain criteria reached. One of the suggestion for the caption is to implement your work as follow:<br>
    <span style="color:white;">........</span>a loop when true<br>
<span style="color:white;">..............</span>ask user whether there are any more processes<br>
<span style="color:white;">..............</span>if no then<br>
<span style="color:white;">...................</span>terminate the loop<br>
<span style="color:white;">..............</span>otherwise<br>
<span style="color:white;">...................</span>do the process
</p>

-----------------------------------

In [2]:
## Repeat the calculation of percentage markup and profit margin
## until a certain criteria reached.

while True:
    proceed = input("\nContinue the process (y/n): ")
    if (proceed == "N" or proceed == "n"):
    ....



Continue the process (y/n): y
Please enter the product's selling price : 834
Please enter the product's purchase price: 430

The selling price           : $    834.00
The purchase price          : $    430.00
The gross profit            : $    404.00
The percentage gross profit :       93.95%
The profit margin           :       48.44%

Continue the process (y/n): n
Work completed!!


---------------------

<h3 style="color:blue;font-size=11px"><b>Question #3</b></h3>
<p style="color:DodgerBlue;font-size:12px">A men’s clothing store advertises that if you buy a suit, you can get a second suit at 50% off. What they mean is that if you buy two suits, then the price of the lower-cost suit is reduced by 50%. Write a program that accepts the two costs as input and then calculates the total cost after halving the cost of the lowest price suit. In addition, a sales tax of 7% applied on the amount of purchased that rounding down to the nearest dollars. For example, if the calculated sales tax is \$0.86, the payable sales tax is \$0.00; if the calculated sales tax is \$11.67, the payable sales tax is \$11.00.</p>

------------------------

In [3]:
## The main program - i.e. control body of question #3
def q3Main():
    ## Input the purchase
    prices = ....             # prices is a list of values return from a function
    
    if len(prices) != 0:
        ## Display the purchases
        ....
        
        ## Determin the discounts on every even purchases
        ....
        
        ## Determine tax of purchase
        ....
        
        ## Display the purchase invoice
        ....
        
    
## Funtion used to input the purhases (i.e. purchases of more than 1 items)    
def inputPurchase():
    ....
    return (....)          # return a number of purchases as a list


## Function to determine the discounts on every even items
def amountDiscount(....):    # pass the list of purchase via parameter
    ....
    return (amount, discount)        # return the amount of purchase and discount


## Function to determine the tax payable on purchases after discount
def determineTax(amt, dis):        # pass the amount of purchase and discount
    ....
    return (....)        # return the taxpayable on amount purchase after discount


## Function to display the purchases according to the sequence of inputs
def displayPurchase(....):        # pass list of purchase to display
    ....
        

## Function to display the various amount of the invoice        
def displayInvoice(....):     # pass all relevant values to display via parameter
    ....


## Call statement to activiate the program
q3Main()
    

Enter the price of purchase (0 to terminate input): 8375
Enter the price of purchase (0 to terminate input): 2569
Enter the price of purchase (0 to terminate input): 3678
Enter the price of purchase (0 to terminate input): 4280
Enter the price of purchase (0 to terminate input): 0

Invoice of purchase
Item 1 price                      :   8,375.00
Item 2 price                      :   2,569.00
Item 3 price                      :   3,678.00
Item 4 price                      :   4,280.00

Amount of purchase                :  18,902.00
Discount                          :   3,424.50
Amount of purchase after discount :  15,477.50
Tax on purchase after discount    :   1,083.43
Amount charged                    :  16,560.93


-------------------------

<h3 style="color:blue;"><b>Question #4</b></h3>
<p style="color:DodgerBlue;font-size:12px;">Write a program that determine the final scores and its corresponding letter grade according to the followings:	(30 marks)</p>
<p style="color:SlateBlue;">Assessment method:<br>
    <table style="color:SlateBlue;">
        <tr>
            <td style="width:200px; text-align:left;">In-class exercises</td>
            <td style="width:80px;">10%</td>
        </tr>
        <tr>
            <td style="width:200px; text-align:left;">Assignments</td>
            <td style="width:80px;">20%</td>
        </tr>
        <tr>
            <td style="width:200px; text-align:left;">Mid-term</td>
            <td style="width:80px;">30%</td>
        </tr>
        <tr>
            <td style="width:200px; text-align:left;">Exam</td>
            <td style="width:80px;">40%</td>
        </tr>
    </table></p>
<p style="color:SlateBlue;">Grading method:<br>
    <table style="color:SlateBlue;">
        <tr>
            <td style="width:200px; text-align:left;">90 ~ 100</td>
            <td style="width:80px;">A</td>
        </tr>
        <tr>
            <td style="width:200px; text-align:left;">80 ~ 89.9</td>
            <td style="width:80px;">B</td>
        </tr>        
        <tr>
            <td style="width:200px; text-align:left;">70 ~ 79.9</td>
            <td style="width:80px;">C</td>
        </tr>
        <tr>
            <td style="width:200px; text-align:left;">60 ~ 69.9</td>
            <td style="width:80px;">D</td>
        </tr>
        <tr>
            <td style="width:200px; text-align:left;">50 ~ 59.9</td>
            <td style="width:80px;">E</td>
        </tr>
        <tr>
            <td style="width:200px; text-align:left;">0 ~ 49.9</td>
            <td style="width:80px;">F</td>
        </tr>        
        </table></p>        
<p  style="color:DodgerBlue;font-size:12px;">The input from the user will be in one single line as follow:<br>
bb912345 56 78 69 80</p>

-----------------------------------

In [4]:
## main program - control body of question #4
def q4Main():
    ## Capture inputs
    ....
    
    ## If lInput not equal to [] proceed
    if .......:        # if list of input is not []
        
        ## Determine the final scores
        ....
        
        ## Determine the corresponding letter grade
        ....
        
        ## Display the academic report accordingly
        ....
            

## function to capture the inputs and return the input as a list            
def captureInputs():
    ....
    return ....          # return the inputs as a list of items    


## Function to determine the final score according t0 the assessment formula
def finalScores(....):   # pass the list of items via parameter
    ....
    return ....          # return the calculated final scores


## Function to determine the final grade according to the grading scheme
def finalGrade(pts):     # pass the final scores via parameter
    ....
    return ....          # return the corresponding final grade
    
    
## Function to display the academic report of a student according to the inputs    
def displayReport(...):    # pass the various value for display as academic report
    ....

    
## The call statement to activiate the program
q4Main()


Please enter ID and scores: bb912345 86 74 56 92

Student:                BB912345
In-class exercises:     86
Assignments:            74
Mid-term:               56
Exam:                   92
Overall scores:         77
Overall grade:          C
